File: app/src/main/java/com/timper/lonelysword/app/feature/main/MainFragment.java
//package com.timper.lonelysword.app.feature.main;
//
//import android.databinding.ObservableField;
//import android.widget.Toast;
//import com.timper.lonelysword.annotations.apt.AfterViews;
//import com.timper.lonelysword.annotations.apt.BeforeViews;
//import com.timper.lonelysword.annotations.apt.DisableNetwork;
//import com.timper.lonelysword.annotations.apt.EnableNetwork;
//import com.timper.lonelysword.annotations.apt.ModelAdapter;
//import com.timper.lonelysword.annotations.apt.RootView;
//import com.timper.lonelysword.annotations.apt.ViewModel;
//import com.timper.lonelysword.annotations.aspectj.CheckLogin;
//import com.timper.lonelysword.annotations.aspectj.SingleClick;
//import com.timper.lonelysword.annotations.aspectj.Time;
//import com.timper.lonelysword.app.R;
//import com.timper.lonelysword.app.databinding.FrgMainBinding;
//import com.timper.lonelysword.app.feature.MainModelAdapter;
//import com.timper.lonelysword.app.feature.MainViewModel;
//import com.timper.lonelysword.base.AppFragment;
//import com.timper.lonelysword.base.ModelAdapterFactor;
//import javax.inject.Inject;
//
///**
// * User: tangpeng.yang
// * Date: 31/05/2018
// * Description:
// * FIXME
// */
//@ViewModel @RootView(R.layout.frg_main) public class MainFragment extends AppFragment<FrgMainBinding> {
//
//  @Inject ModelAdapterFactor<MainModelAdapter> factor;
//  @ModelAdapter("factor") MainModelAdapter adapter;
//
//  public final ObservableField<String> abc = new ObservableField<>("SDFASDFSADFSADF");
//  //@Inject @ViewModel MainViewModel viewModel;
//
//  @BeforeViews void beforViews() {
//    Toast.makeText(getActivity(), "beforviews", Toast.LENGTH_LONG).show();
//  }
//
//  @AfterViews void AfterViews() {
//
//    Toast.makeText(getActivity(), "AfterViews", Toast.LENGTH_LONG).show();
//  }
//
//  @DisableNetwork void disable() {
//    Toast.makeText(getActivity(), "disable", Toast.LENGTH_LONG).show();
//  }
//
//  @EnableNetwork @CheckLogin void enable() {
//    Toast.makeText(getActivity(), "enable", Toast.LENGTH_LONG).show();
//  }
//
//  @SingleClick @Time void click() {
//    Toast.makeText(getActivity(), "SingleClick", Toast.LENGTH_SHORT).show();
//  }
//}

File: demo/module/src/main/java/com/timper/module/feature/garden/main/MainFragment.kt
package com.timper.module.feature.garden.main

import com.timper.bindingadapter.collection.itembindings.OnItemBindClass
import com.timper.bindingadapter.collection.itembindings.OnItemBindEvent
import com.timper.lonelysword.annotations.apt.AfterViews
import com.timper.lonelysword.annotations.apt.Dagger
import com.timper.lonelysword.annotations.apt.RootView
import com.timper.lonelysword.base.AppFragment
import com.timper.module.BR
import com.timper.module.R
import com.timper.module.R2
import com.timper.module.data.bean.Article
import com.timper.module.databinding.FrgMainBinding
import com.timper.module.feature.garden.GardenActivity

@Dagger(GardenActivity::class)
@RootView(R2.layout.frg_main)
class MainFragment : AppFragment<MainViewModel,FrgMainBinding>() {

    /**
     * drawerBinding
     */
    var itemBinding: OnItemBindClass<Article> =
        OnItemBindClass<Article>().map(Article::class.java, OnItemBindEvent<Article>(BR.viewModel, R.layout.item_main))

    @AfterViews
    internal fun afterViews() {
    }

    companion object {
        fun instance(): MainFragment {
            return MainFragment()
        }
    }
}
